### ISSUE TITLE
Can't test Eloquent relationships without hitting the DB

### ISSUE THREAD

**Issue by hackel** [BenConstable/phpspec-laravel#29]
All of my model tests just stopped working with a recent update.  Now they try to hit the DB, and when they can't connect, it causes my specs to fail.

I am testing it like this:
`$this->categories()->shouldDefineRelationship('belongsToMany', 'Models\Category');`
Using the latest revision ce6379d9d7910959830fa6260b41a46acff44ddc.

**Comment by BenConstable** [comment]
I had a good look into this, and I can't see a way of stopping Laravel from requiring a database connection when running Eloquent relationship methods.

Rather than leaving it broken or requiring users to have a booted database, I've removed the custom matcher (and Eloquent functionality). I've added a [note in the README](https://github.com/BenConstable/phpspec-laravel#a-note-on-the-database-and-eloquent) to reflect this.

I'm going to leave this issue open for now - so if anyone can figure it out and make PR it would be greatly appreciated!
